git_machete/client.py

```diff
@@ -1182,6 +1182,23 @@ def delete_unmanaged(self, *, opt_yes: bool) -> None:
         branches_to_delete = excluding(self.__git.get_local_branches(), self.managed_branches)
         self.__delete_branches(branches_to_delete=branches_to_delete, opt_yes=opt_yes)
 
+    def __delete_branches_rearrange_layout(self, branches_to_delete: List[LocalBranchShortName], opt_yes: bool) -> None:
```

Comment:
Hmmm AFAICS this command first slides out, then deletes the branches... how about packing that functionality into `slide-out --merged [--delete]`? I can see the following benefits:

1. Consistent behavior &mdash; slide-out serves the purpose of removing the branches from `.git/machete`; then whether to delete them can be controlled with `--delete` flag
2. Actually, deleting branches right after sliding out, while ofc possible with `--delete` flag (let alone with `git branch -d`), is kind of discouraged due to the reasons outlined in [FAQ](https://github.com/VirtusLab/git-machete#sometimes-when-i-run-update-or-traverse-too-many-commits-are-taken-into-the-rebase-how-to-fix-that). So better not make it the default behavior straight away 😅 
3. Arguably less CLI bloat 😅 git-machete got already has (too) many top-level commands, it feels better to keep related stuff in long-established commands like `slide-out` where the users can expected them,

Reply:
one awkward thing about `slide-out --merged` is, if you don't pass `--delete`, it will remove the branch from the layout. if you realize you wanted the branches deleted, you can't immediately `slide-out --merged --delete` because those branches are now unmanaged

to automatically find and delete them at this point you probably have to `discover` and then `slide-out --merged --delete` or something

Reply:
> to automatically find and delete them at this point you probably 

Note that there's `git machete delete-unmanaged` (already mentioned in this conversation)... and there's also a long-standing task (https://github.com/VirtusLab/git-machete/issues/98) to implement support for `git config machete.slideOut.deleteLocalBranch true` that would delete branches on slide-out by default. That's actually one of the few features present in [git-machete-intellij-plugin](https://github.com/VirtusLab/git-machete-intellij-plugin) but not in this CLI 😅 

So yeah, if you could proceed with `slide-out --merged`... I can smooth out all the details like shell completions etc. at the end ✍🏻

Reply:
I went with `slide-out --all-merged` since there's already a `slide-out --merge` flag

it's... kinda awkward though. there are a bunch of other args to slide-out that don't make any sense. the whole pass a chain thing doesn't apply here either

Reply:
I'm gonna put in in docs as another flavor of slide-out, like

```
    git machete slide-out --all-merged [--delete]
    git machete slide-out [-d|--down-fork-point=<down-fork-point-commit>] [--delete] [-M|--merge] [-n|--no-edit-merge|--no-interactive-rebase] [<branch> [<branch> [<branch> ...]]]
```

similarly to `fork-point` which has a couple flavors as well